From 242196b0c77841bd55881c06f8c71d251acc351a Mon Sep 17 00:00:00 2001
From: Kiril Mikos <kiril.mik.os@gmail.com>
Date: Wed, 24 Oct 2012 21:48:07 +0300
Subject: [PATCH 08/28] LatinIME: Apply "Enable more keys for top row on
 tablets" (f6620308ba) and "Add "more" keys to ALL the keys" (c561f31af9) to
 slavik layouts.

Change-Id: Ic6fea3faecc5783d99f5257a4501e37b79f40fdf
---
 java/res/xml/rowkeys_east_slavic2.xml | 40 +++++++++++++++++++++------
 java/res/xml/rowkeys_east_slavic3.xml | 36 ++++++++++++++++++------
 2 files changed, 58 insertions(+), 18 deletions(-)

diff --git a/java/res/xml/rowkeys_east_slavic2.xml b/java/res/xml/rowkeys_east_slavic2.xml
index 21463fb7d..cb8558ff4 100644
--- a/java/res/xml/rowkeys_east_slavic2.xml
+++ b/java/res/xml/rowkeys_east_slavic2.xml
@@ -23,37 +23,59 @@
 >
     <!-- U+0444: "ф" CYRILLIC SMALL LETTER EF -->
     <Key
-        latin:keySpec="&#x0444;" />
+        latin:keySpec="&#x0444;"
+        latin:keyHintLabel="\@"
+        latin:additionalMoreKeys="\\@" />
     <Key
         latin:keySpec="!text/keyspec_east_slavic_row2_2"
+        latin:keyHintLabel="#"
+        latin:additionalMoreKeys="#"
         latin:moreKeys="!text/morekeys_east_slavic_row2_2" />
     <!-- U+0432: "в" CYRILLIC SMALL LETTER VE -->
     <Key
-        latin:keySpec="&#x0432;" />
+        latin:keySpec="&#x0432;"
+        latin:keyHintLabel="$"
+        latin:additionalMoreKeys="$" />
     <!-- U+0430: "а" CYRILLIC SMALL LETTER A -->
     <Key
         latin:keySpec="&#x0430;"
-        latin:moreKeys="!text/morekeys_cyrillic_a" />
+        latin:moreKeys="!text/morekeys_cyrillic_a"
+        latin:keyHintLabel="%"
+        latin:additionalMoreKeys="%" />
     <!-- U+043F: "п" CYRILLIC SMALL LETTER PE -->
     <Key
-        latin:keySpec="&#x043F;" />
+        latin:keySpec="&#x043F;"
+        latin:keyHintLabel="&amp;"
+        latin:additionalMoreKeys="&amp;" />
     <!-- U+0440: "р" CYRILLIC SMALL LETTER ER -->
     <Key
-        latin:keySpec="&#x0440;" />
+        latin:keySpec="&#x0440;"
+        latin:keyHintLabel="*"
+        latin:additionalMoreKeys="*" />
     <!-- U+043E: "о" CYRILLIC SMALL LETTER O -->
     <Key
         latin:keySpec="&#x043E;"
+        latin:keyHintLabel="-"
+        latin:additionalMoreKeys="-"
         latin:moreKeys="!text/morekeys_cyrillic_o" />
     <!-- U+043B: "л" CYRILLIC SMALL LETTER EL -->
     <Key
-        latin:keySpec="&#x043B;" />
+        latin:keySpec="&#x043B;"
+        latin:keyHintLabel="+"
+        latin:additionalMoreKeys="+" />
     <!-- U+0434: "д" CYRILLIC SMALL LETTER DE -->
     <Key
-        latin:keySpec="&#x0434;" />
+        latin:keySpec="&#x0434;"
+        latin:keyHintLabel="_"
+        latin:additionalMoreKeys="_" />
     <!-- U+0436: "ж" CYRILLIC SMALL LETTER ZHE -->
     <Key
-        latin:keySpec="&#x0436;" />
+        latin:keySpec="&#x0436;"
+        latin:keyHintLabel="("
+        latin:additionalMoreKeys="(" />
     <Key
         latin:keySpec="!text/keyspec_east_slavic_row2_11"
-        latin:moreKeys="!text/morekeys_east_slavic_row2_11" />
+        latin:moreKeys="!text/morekeys_east_slavic_row2_11"
+        latin:keyHintLabel=")"
+        latin:additionalMoreKeys=")" />
 </merge>
diff --git a/java/res/xml/rowkeys_east_slavic3.xml b/java/res/xml/rowkeys_east_slavic3.xml
index 54802e878..77151ffdb 100644
--- a/java/res/xml/rowkeys_east_slavic3.xml
+++ b/java/res/xml/rowkeys_east_slavic3.xml
@@ -23,29 +23,47 @@
 >
     <!-- U+044F: "я" CYRILLIC SMALL LETTER YA -->
     <Key
-        latin:keySpec="&#x044F;" />
+        latin:keySpec="&#x044F;"
+        latin:keyHintLabel="!"
+        latin:additionalMoreKeys="!" />
     <!-- U+0447: "ч" CYRILLIC SMALL LETTER CHE -->
     <Key
-        latin:keySpec="&#x0447;" />
+        latin:keySpec="&#x0447;"
+        latin:keyHintLabel="&quot;"
+        latin:additionalMoreKeys="&quot;" />
     <!-- U+0441: "с" CYRILLIC SMALL LETTER ES -->
     <Key
-        latin:keySpec="&#x0441;" />
+        latin:keySpec="&#x0441;"
+        latin:keyHintLabel="&apos;"
+        latin:additionalMoreKeys="&apos;" />
     <!-- U+043C: "м" CYRILLIC SMALL LETTER EM -->
     <Key
-        latin:keySpec="&#x043C;" />
+        latin:keySpec="&#x043C;"
+        latin:keyHintLabel=":"
+        latin:additionalMoreKeys=":" />
     <Key
-        latin:keySpec="!text/keyspec_east_slavic_row3_5" />
+        latin:keySpec="!text/keyspec_east_slavic_row3_5"
+        latin:keyHintLabel=";"
+        latin:additionalMoreKeys=";" />
     <!-- U+0442: "т" CYRILLIC SMALL LETTER TE -->
     <Key
-        latin:keySpec="&#x0442;" />
+        latin:keySpec="&#x0442;"
+        latin:keyHintLabel="/"
+        latin:additionalMoreKeys="/" />
     <!-- U+044C: "ь" CYRILLIC SMALL LETTER SOFT SIGN -->
     <Key
         latin:keySpec="&#x044C;"
-        latin:moreKeys="!text/morekeys_cyrillic_soft_sign" />
+        latin:moreKeys="!text/morekeys_cyrillic_soft_sign"
+        latin:keyHintLabel="\?"
+        latin:additionalMoreKeys="\\?" />
     <!-- U+0431: "б" CYRILLIC SMALL LETTER BE -->
     <Key
-        latin:keySpec="&#x0431;" />
+        latin:keySpec="&#x0431;"
+        latin:keyHintLabel="&lt;"
+        latin:additionalMoreKeys="&lt;" />
     <!-- U+044E: "ю" CYRILLIC SMALL LETTER YU -->
     <Key
-        latin:keySpec="&#x044E;" />
+        latin:keySpec="&#x044E;"
+        latin:keyHintLabel="&gt;"
+        latin:additionalMoreKeys="&gt;" />
 </merge>
-- 
2.17.1

